x-pack/plugins/enterprise_search/public/applications/app_search/types.ts

```diff
@@ -6,10 +6,4 @@
 
 export * from '../../../common/types/app_search';
 export { IRole, TRole, TAbility } from './utils/role';
-
-export interface IEngine {
-  name: string;
-  type: string;
-  language: string;
-  result_fields: object[];
```

Comment:
This is an example of the fallibility of Typescript definitions, especially when handling server-side data. We've been using an incorrect type for over a year now. `result_fields` is an object, **not** an array of objects:

<img width="232" alt="Screen Shot 2020-11-09 at 10 32 35 AM" src="https://user-images.githubusercontent.com/549407/98582001-2f015080-2277-11eb-89b0-c52761d38372.png">

But because we've mostly passed `result_fields` through and never operated very specifically on it, we haven't caught it until now.

Reply:
Woah, that's wild.